swift/ios-shoppe-demo/ServiceManagers/FullStoryManager.swift

```diff
@@ -97,3 +98,57 @@ func fsModifyPrivacy(setting: PrivacySetting, views: UIView?...) {
 func fsIdentify(id: String, userInfo: [String: Any]) {
     FS.identify(id, userVars: userInfo)
 }
+
+
+// below are functions for demostrating conversions: revenue attribution events
+func makeId(len: Int) -> String {
+    let letters : NSString = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+    let length = UInt32(letters.length)
+
+    var randomString = ""
+
+    for _ in 0 ..< len {
+        let rand = arc4random_uniform(length)
+        var nextChar = letters.character(at: Int(rand))
+        randomString += NSString(characters: &nextChar, length: 1) as String
+    }
+
+    return randomString
```

Comment:
The Swift standard library now has randomization APIs. So this can be simplified:

```suggestion
    let characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"

    var randomString = ""

    for _ in 0 ..< len {
        let randomCharacter = characters.randomElement()!
        randomString.append(randomCharacter)
    }

    return randomString
```